src/lib/parse.ts

```diff
@@ -0,0 +1,48 @@
+import { ParseError, UnsupportedFeatureError, WrongInputError } from "~/errors";
+import { AllowedBumps } from "~/types";
+
+function getName(title: string): string {
+  const match = /(bump|update) (?<name>(?:@[^\s]+\/)?[^\s]+) (requirement)?/i.exec(title);
+
+  if (match && match.groups && match.groups.name) {
+    return match.groups.name;
+  }
+
+  throw new ParseError("No valid dependancy 'name' found in PR title.");
+}
+
+function getRawVersion(title: string, target: "from" | "to"): string {
+  const regex =
+    /(?<version>(?<major>0|[1-9]\d*)(\.(?<minor>(0|[1-9]\d*))(\.(?<patch>(0|[1-9]\d*))(?:-((?<remainder>0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)?)?)/;
```

Comment:
😄
- `major`, `minor` and `patch` can be numbers, from 0 to (basically) infinite
- the whole is grouped under version.
- `major` is mandatory
- `minor` is optional but...
- mandatory if `patch` is present
- `patch` is optional but...
- mandatory if `remainder` is present
- `remainder` is optional atm because we just don't support automerge for something that would be in prerelease like `1.1.1-alpha.1`. For now, it's only used as kind of security. If `patch && !remainder` we handle it, if `patch && remainder` -> `UnsupportedError`

Reply:
Good idea, I added it 👍